src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs

```diff
@@ -267,5 +269,48 @@ static void Main()
                 });
             }
         }
+
+        [WorkItem(42361, "https://github.com/dotnet/roslyn/issues/42361")]
+        [ConditionalFact(typeof(WindowsOnly))]
+        public void Constraints()
+        {
+            int n = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
+            {
+                (ExecutionArchitecture.x86, ExecutionConfiguration.Debug) => 420,
```

Comment:
Is this change fixing a regression in our ability to handle complex constraint graphs? Or improving our ability past where it has historically been?

Reply:
The change improves the ability to handle complex constraint graphs, although 15.9 uses less stack space still.

---
In reply to: [405771709](https://github.com/dotnet/roslyn/pull/43094#discussion_r405771709) [](ancestors = 405771709)